### PR TITLE
Change strings with .format() calls to f-strings in color, exposure, transform, and util subpackages

### PR DESCRIPTION
**Pull request by MariosAchilias**
## Description

 Partially addresses #5474

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.

### CODE REVIEW

File: skimage/color/colorconv.py [hunk 1]
```diff
@@ -535,8 +535,8 @@ def get_xyz_coords(illuminant, observer, dtype=float):
     try:
         return np.asarray(illuminants[illuminant][observer], dtype=dtype)
     except KeyError:
-        raise ValueError("Unknown illuminant/observer combination\
-        (\'{0}\', \'{1}\')".format(illuminant, observer))
+        raise ValueError(f"Unknown illuminant/observer combination"
+                         f"(\'{illuminant}\', \'{observer}\')")
```
Comment (mkcor) [inline skimage/color/colorconv.py:539]: I guess there's no need to escape the single quotes, is there?

File: skimage/transform/radon_transform.py [hunk 2]
```diff
@@ -485,8 +485,8 @@ def iradon_sart(radon_image, theta=None, image=None, projection_shifts=None,
                          'of radon_image (%s)'
                          % (image.shape, reconstructed_shape))
     elif image.dtype != dtype:
-        warn("image dtype does not match output dtype: "
-             "image is cast to {}".format(dtype))
+        warn(f"image dtype does not match output dtype: "
+             f"image is cast to {dtype}")
```
Comment (mkcor) [inline skimage/transform/radon_transform.py:489]: Single quotes would be enough in this case.

File: skimage/util/dtype.py [hunk 3]
```diff
@@ -130,12 +130,12 @@ def _scale(a, n, m, copy=True):
     if n > m and a.max() < 2 ** m:
         mnew = int(np.ceil(m / 2) * 2)
         if mnew > m:
-            dtype = "int{}".format(mnew)
+            dtype = f"int{mnew}"
```
Comment (mkcor) [inline skimage/util/dtype.py:133]: Same here, and in the following.

File: skimage/util/dtype.py [hunk 4]
```diff
@@ -252,8 +252,8 @@ def _convert(image, dtype, force_copy=False, uniform=False):
         return image
 
     if not (dtype_in in _supported_types and dtype_out in _supported_types):
-        raise ValueError("Can not convert from {} to {}."
-                         .format(dtypeobj_in, dtypeobj_out))
+        raise ValueError(f"Can not convert from "
+                         f"{dtypeobj_in} to {dtypeobj_out}.")
```
Comment (mkcor) [inline skimage/util/dtype.py:256]:
```suggestion
        raise ValueError(f'Cannot convert from {dtypeobj_in} to '
                         f'{dtypeobj_out}.')
```

**Comment by MariosAchilias** [comment]
I made the suggested changes. Thank you for the review.

Is there anything else I should change?

File: skimage/util/dtype.py [hunk 5]
```diff
@@ -252,8 +252,8 @@ def _convert(image, dtype, force_copy=False, uniform=False):
         return image
 
     if not (dtype_in in _supported_types and dtype_out in _supported_types):
-        raise ValueError("Can not convert from {} to {}."
-                         .format(dtypeobj_in, dtypeobj_out))
+        raise ValueError(f'Can not convert from {dtypeobj_in} to '
```
Comment (grlee77) [inline skimage/util/dtype.py:255]:
```suggestion
        raise ValueError(f'Cannot convert from {dtypeobj_in} to '
```

unrelated to f-strings, but I think "cannot" is more common usage than "can not", although I think either is technically correct and both are more formal than `can't`

**Review by grlee77 — CHANGES_REQUESTED** [review]
The changes look good to me, but it looks like a couple files in these modules were missed (found by running `grep -r ".format("` in the skimage folder):
`skimage/transform/_warps.py`
`skimage/color/tests/test_colorconv.py`

Can you please update those two here as well? Thanks

File: skimage/util/dtype.py (hunk 5 above)
Comment (mkcor) [reply skimage/util/dtype.py:255]: Yes, [indeed](https://www.merriam-webster.com/words-at-play/cannot-vs-can-not-is-there-a-difference). I had already made the [suggestion](https://github.com/scikit-image/scikit-image/pull/5508#discussion_r686933106) (and @MariosAchilias +1'd)), but it looks like it fell in the cracks in 774dc0c.

**Review by grlee77 — APPROVED** [review]

**Comment by grlee77** [comment]
Thanks @MariosAchilias!

**Comment by MariosAchilias** [comment]
@grlee77 @mkcor Again, thank you for the reviews :)

This is actually my first(ever!) PR.

**Pull request merged**